### PR TITLE
Add new datasource, update func, update to 0.4.2

### PR DESCRIPTION
**Pull request by tr0njavolta**
This PR updates the provider:
-  to plugin framework version 0.4.2
- creates an `_order` datasource.
- Adds the update function modified to accommodate the hashicups API /orders

### CODE REVIEW

File: hashicups/resource_order.go [hunk 1]
```diff
@@ -259,14 +272,14 @@ func (r resourceOrder) Update(ctx context.Context, req tfsdk.UpdateResourceReque
...
+	// Wrap the results from the order in the Terraform schema with the items and new computed values
+	result := Order{
 		ID:          types.String{Value: strconv.Itoa(order.ID)},
 		Items:       ois,
 		LastUpdated: types.String{Value: string(time.Now().Format(time.RFC850))},
 	}
 
-	// Set state
+	// Write the response result variable to state
 	diags = resp.State.Set(ctx, result)
```
Comment (im2nguyen) [inline hashicups/resource_order.go:283]: showing my ignorance here.. but how come for this, we can set it directly to result, but for the others we do `& resourceData`. is there a difference? which one is preferable?

**Review by im2nguyen — APPROVED** [review]
had a question about `resp.State.Set` but this looks solid

great job! really like the comments you added to guide users through this 😄

File: hashicups/resource_order.go (hunk 1 above)
Comment (im2nguyen) [reply hashicups/resource_order.go:283]: Thanks for the explanation!

**Comment by judithpatudith** [comment]
@tr0njavolta does this PR still need anything? It's been open for a while and maybe we can close it if it's stale?

**Pull request closed without merging**